src/mpid/ch4/src/mpidig_recv_utils.h

```diff
@@ -33,6 +33,10 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_recv_check_rndv_cb(MPIR_Request * rreq)
     int mpi_errno = MPI_SUCCESS;
 
     if (MPIDIG_REQUEST(rreq, req->recv_async).data_copy_cb) {
+        /* the callback may complete the request. Increment the cc to prevent
+         * rreq being freed */
+        MPIR_cc_inc(rreq->cc_ptr);
```

Comment:
Typically, I think, we do this with an additional reference rather than with the completion counter. Would that work?

Reply:
Yes, I think so.